src/polygon/src/polygon.cpp

```diff
@@ -1074,9 +1077,9 @@ std::vector<glm::ivec3> TriangulateIdx(const PolygonsIdx &polys,
     Monotones monotones(polys, precision);
     monotones.Triangulate(triangles);
 #ifdef MANIFOLD_DEBUG
-    if (params.intermediateChecks) {
+    if (kCheckTriangulation) {
       CheckTopology(triangles, polys);
-      if (!params.processOverlaps) {
```

Comment:
I am completely baffled as to what is going on with these `params`, but as far as I can tell they are *very* unreliable - but work just often enough to make me think I wrote it properly. I was finding the weirdest behavior, like I could check the value of `PolygonParams().intermediateChecks` in a `TEST` and see it was true both before and after a Boolean op, yet inside the op, the triangulator would somehow still say `params.intermediateChecks` was false, despite nothing writing to it. I know globals are problematic, but this is downright disturbing. I'd love to know what I'm doing wrong, because it would be really nice to be able to change these settings for different tests.

Reply:
This is weird, I don't think you did anything wrong here. For `params.processOverlaps` I think it works as expected.

Do you have a specific example where it does not work so I can have a look at?

Reply:
You can also use gdb data watchpoint to see what is writing to a certain address. https://sourceware.org/gdb/onlinedocs/gdb/Set-Watchpoints.html

Reply:
Okay, I put a simple repro in #510. I won't get back to this for 12 hours or so (if then). Would love some insight on this!